typing/typeclass.ml

```diff
@@ -1620,23 +1637,34 @@ let final_decl env define_class
       in
       raise(Error(cl.pci_loc, env, Unbound_type_var(printer, reason)))
   end;
-
-  (id, cl.pci_name, clty, ty_id, cltydef, obj_id, obj_abbr, cl_id, cl_abbr,
-   arity, pub_meths, coe, expr,
-   { ci_loc = cl.pci_loc;
-     ci_virt = cl.pci_virt;
-     ci_params = ci_params;
-(* TODO : check that we have the correct use of identifiers *)
-     ci_id_name = cl.pci_name;
-     ci_id_class = id;
-     ci_id_class_type = ty_id;
-     ci_id_object = obj_id;
-     ci_id_typehash = cl_id;
-     ci_expr = expr;
-     ci_decl = clty;
-     ci_type_decl = cltydef;
-     ci_attributes = cl.pci_attributes;
- })
+  { id;
+    id_loc = cl.pci_name;
+    clty;
+    ty_id;
+    cltydef;
+    obj_id;
+    obj_abbr;
+    cl_id;
+    cl_abbr;
+    arity;
+    pub_meths;
+    coe;
+    expr;
```

Comment:
A style suggestion, now that punning is in place, would be to put all the punned identifiers together and reduce the line count):

```ocaml
   { id; clty; ty_id; cltydef; obj_id; obj_abbr; cl_id; cl_abbr; arity;
     pub_meths; coe; expr;
     id_loc = cl.pci_name;
```

(to me it draws more attention to the fields which require expressions - the type checker has your back for any missing fields, of course)